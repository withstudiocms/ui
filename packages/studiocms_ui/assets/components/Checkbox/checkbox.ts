const elements = document.querySelectorAll<HTMLDivElement>('.sui-checkmark-container');
const checkbox = document.querySelectorAll<HTMLInputElement>('.sui-checkbox');

for (const element of elements) {
  if (element.dataset.initialized) continue;
  
  element.dataset.initialized = 'true';

  element.addEventListener('keydown', (e) => {
    if (e.key !== 'Enter' && e.key !== ' ') return;

    e.preventDefault();

    const checkbox = element.querySelector<HTMLInputElement>('.sui-checkbox');
      
    if (!checkbox) return;

    checkbox.click();
  });
}

for (const box of checkbox) {
  if (box.dataset.initialized) continue;

  box.dataset.initialized = 'true';

  box.addEventListener('change', (e) => {
    box.parentElement!.ariaChecked = (e.target as HTMLInputElement).checked ? 'true' : 'false';
  });
}
