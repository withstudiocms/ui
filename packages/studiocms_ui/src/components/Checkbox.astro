---
import Checkmark from '../icons/Checkmark.astro';
import type { StudioCMSColorway } from '../utils/colors';
import { generateID } from '../utils/generateID';

/**
 * The props for the Checkbox component.
 */
interface Props {
  /**
   * The label of the checkbox.
   */
	label: string;
  /**
   * The size of the checkbox. Defaults to `md`.
   */
	size?: 'sm' | 'md' | 'lg';
  /**
   * The color of the checkbox. Defaults to `default`.
   */
	color?: StudioCMSColorway;
  /**
   * Whether the checkbox is checked by default.
   */
	defaultChecked?: boolean;
  /**
   * Whether the checkbox is disabled.
   */
	disabled?: boolean;
  /**
   * The name of the checkbox.
   */
	name?: string;
  /**
   * Whether the checkbox is required.
   */
	isRequired?: boolean;
}

const {
	size = 'md',
	color = 'default',
	defaultChecked,
	disabled,
	name = generateID('checkbox'),
	label,
	isRequired,
} = Astro.props;

const iconSizes = {
	sm: 14,
	md: 16,
	lg: 24,
};
---
<label 
  class="sui-checkmark-label"
  for={name}
  class:list={[
    disabled && "disabled",
    color,
    size,
  ]}
>
  <div
    class="sui-checkmark-container"
    tabindex="0"
    role="checkbox"
    aria-checked={defaultChecked} 
    aria-labelledby={`label-${name}`}
    >
    <Checkmark
      class={'sui-checkmark'}
      width={iconSizes[size]}
      height={iconSizes[size]}
    />
    <input
      type="checkbox"
      name={name}
      id={name}
      checked={defaultChecked}
      disabled={disabled}
      required={isRequired}
      class="sui-checkbox"
      hidden
    />
  </div>
  <span id={`label-${name}`}>
    {label} <span class="req-star">{isRequired && "*"}</span>
  </span>
</label>
<script>
  const elements = document.querySelectorAll<HTMLDivElement>('.sui-checkmark-container');
  const checkbox = document.querySelectorAll<HTMLInputElement>('.sui-checkbox');

  for (const element of elements) {
    if (element.dataset.initialized) continue;
    
    element.dataset.initialized = 'true';

    element.addEventListener('keydown', (e) => {
      if (e.key !== 'Enter' && e.key !== ' ') return;

      e.preventDefault();

      const checkbox = element.querySelector<HTMLInputElement>('.sui-checkbox');
        
      if (!checkbox) return;

      checkbox.click();
    });
  }

  for (const box of checkbox) {
    if (box.dataset.initialized) continue;

    box.dataset.initialized = 'true';

    box.addEventListener('change', (e) => {
      box.parentElement!.ariaChecked = (e.target as HTMLInputElement).checked ? 'true' : 'false';
    });
  }
</script>
<style>
  .sui-checkmark-label {
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: .5rem;
    position: relative;
    margin: .25rem 0;
  }

  .sui-checkmark-label.disabled {
    opacity: 0.5;
    pointer-events: none;
    color: hsl(var(--text-muted));
  }

  .sui-checkmark-container {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: hsl(var(--default-base));
    border: 2px solid hsl(var(--default-base));
    border-radius: .5rem;
    cursor: pointer;
    transition: background-color .15s, border .15s, transform .15s;
    transition-timing-function: ease;
  }

  .sui-checkmark-container:focus-visible {
    outline: 2px solid hsl(var(--text-normal));
    outline-offset: 2px;
  }

  .sui-checkmark-label:hover .sui-checkmark-container {
    background-color: hsl(var(--default-hover));
  }

  .sui-checkmark-label:active .sui-checkmark-container {
    background-color: hsl(var(--default-active));
    scale: 0.9;
  }

  .sui-checkmark-label.sm {
    font-size: .875em;
  }

  .sui-checkmark-label.sm .sui-checkmark-container {
    width: 1.25rem;
    height: 1.25rem;
  }

  .sui-checkmark-label.md .sui-checkmark-container {
    width: 1.5rem;
    height: 1.5rem;
  }

  .sui-checkmark-label.lg {
    font-size: 1.125em;
  }

  .sui-checkmark-label.lg .sui-checkmark-container {
    width: 1.75rem;
    height: 1.75rem;
    padding: .125rem;
  }

  .primary .sui-checkmark-container:has(.sui-checkbox:checked) {
    border-color: hsl(var(--primary-base));
    background-color: hsl(var(--primary-base));
  }

  .success .sui-checkmark-container:has(.sui-checkbox:checked) {
    border-color: hsl(var(--success-base));
    background-color: hsl(var(--success-base));
  }

  .warning .sui-checkmark-container:has(.sui-checkbox:checked) {
    border-color: hsl(var(--warning-base));
    background-color: hsl(var(--warning-base));
  }

  .danger .sui-checkmark-container:has(.sui-checkbox:checked) {
    border-color: hsl(var(--danger-base));
    background-color: hsl(var(--danger-base));
  }

  .req-star {
    color: hsl(var(--danger-base));
    font-weight: 700;
  }

  .sui-checkbox {
    width: 0;
    height: 0;
    visibility: hidden;
    opacity: 0;
    margin: 0;
  }
</style>
<style is:global>
  .sui-checkmark polyline {
    transition: all .15s ease;
    color: hsl(var(--text-normal));
  }

  .primary .sui-checkmark polyline {
    color: hsl(var(--text-inverted));
  }

  .success .sui-checkmark polyline {
    color: hsl(var(--text-dark));
  }

  .warning .sui-checkmark polyline {
    color: hsl(var(--text-dark));
  }

  .danger .sui-checkmark polyline {
    color: hsl(var(--text-light));
  }

  .sui-checkmark-container:has(.sui-checkbox:checked) {
    .sui-checkmark polyline {
      stroke-dashoffset: 44;
    }
  }
</style>
