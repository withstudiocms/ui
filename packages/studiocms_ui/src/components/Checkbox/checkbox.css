.sui-checkmark-label {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: .5rem;
  position: relative;
  margin: .25rem 0;
}

.sui-checkmark-label.disabled {
  opacity: 0.5;
  pointer-events: none;
  color: hsl(var(--text-muted));
}

.sui-checkmark-container {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: hsl(var(--default-base));
  border: 2px solid hsl(var(--default-base));
  border-radius: var(--radius-md);
  cursor: pointer;
  transition: background-color .15s, border .15s, transform .15s;
  transition-timing-function: ease;
}

.sui-checkmark-container:focus-visible {
  outline: 2px solid hsl(var(--text-normal));
  outline-offset: 2px;
}

.sui-checkmark-label:hover .sui-checkmark-container {
  background-color: hsl(var(--default-hover));
}

.sui-checkmark-label:active .sui-checkmark-container {
  background-color: hsl(var(--default-active));
  scale: 0.9;
}

.sui-checkmark-label.sm {
  font-size: .875em;
}

.sui-checkmark-label.sm .sui-checkmark-container {
  width: 1.25rem;
  height: 1.25rem;
}

.sui-checkmark-label.md .sui-checkmark-container {
  width: 1.5rem;
  height: 1.5rem;
}

.sui-checkmark-label.lg {
  font-size: 1.125em;
}

.sui-checkmark-label.lg .sui-checkmark-container {
  width: 1.75rem;
  height: 1.75rem;
  padding: .125rem;
}

.primary .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--primary-base));
  background-color: hsl(var(--primary-base));
}

.success .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--success-base));
  background-color: hsl(var(--success-base));
}

.warning .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--warning-base));
  background-color: hsl(var(--warning-base));
}

.danger .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--danger-base));
  background-color: hsl(var(--danger-base));
}

.info .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--info-base));
  background-color: hsl(var(--info-base));
}

.mono .sui-checkmark-container:has(.sui-checkbox:checked) {
  border-color: hsl(var(--mono-base));
  background-color: hsl(var(--mono-base));
}

.req-star {
  color: hsl(var(--danger-base));
  font-weight: 700;
}

.sui-checkbox {
  width: 0;
  height: 0;
  visibility: hidden;
  opacity: 0;
  margin: 0;
}

.sui-checkmark polyline {
  transition: all .15s ease;
  color: hsl(var(--text-normal));
}

.primary .sui-checkmark polyline {
  color: hsl(var(--text-inverted));
}

.success .sui-checkmark polyline {
  color: hsl(var(--text-dark));
}

.warning .sui-checkmark polyline {
  color: hsl(var(--text-dark));
}

.danger .sui-checkmark polyline {
  color: hsl(var(--text-light));
}

.info .sui-checkmark polyline {
  color: hsl(var(--text-light));
}

.mono .sui-checkmark polyline {
  color: hsl(var(--text-inverted));
}

.sui-checkmark-container:has(.sui-checkbox:checked) {
  .sui-checkmark polyline {
    stroke-dashoffset: 44;
  }
}
