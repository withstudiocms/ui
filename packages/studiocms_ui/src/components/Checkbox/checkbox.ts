const allElements = document.querySelectorAll<HTMLDivElement>('.sui-checkmark-container');
const allCheckbox = document.querySelectorAll<HTMLInputElement>('.sui-checkbox');

for (const element of allElements) {
	if (element.dataset.initialized) continue;

	element.dataset.initialized = 'true';

	element.addEventListener('keydown', (e) => {
		if (e.key !== 'Enter' && e.key !== ' ') return;

		e.preventDefault();

		const checkbox = element.querySelector<HTMLInputElement>('.sui-checkbox');

		if (!checkbox) return;

		checkbox.click();
	});
}

for (const box of allCheckbox) {
	if (box.dataset.initialized) continue;

	box.dataset.initialized = 'true';

	box.addEventListener('change', (e) => {
		box.parentElement!.ariaChecked = (e.target as HTMLInputElement).checked ? 'true' : 'false';
	});
}
