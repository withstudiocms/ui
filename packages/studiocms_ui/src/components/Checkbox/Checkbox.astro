---
import Checkmark from '../../icons/Checkmark.astro';
import type { StudioCMSColorway } from '../../utils/colors.js';
import { generateID } from '../../utils/generateID.js';
import './checkbox.css';

/**
 * The props for the Checkbox component.
 */
interface Props {
	/**
	 * The label of the checkbox.
	 */
	label: string;
	/**
	 * The size of the checkbox. Defaults to `md`.
	 */
	size?: 'sm' | 'md' | 'lg';
	/**
	 * The color of the checkbox. Defaults to `default`.
	 */
	color?: StudioCMSColorway;
	/**
	 * Whether the checkbox is checked by default.
	 */
	defaultChecked?: boolean;
	/**
	 * Whether the checkbox is disabled.
	 */
	disabled?: boolean;
	/**
	 * The name of the checkbox.
	 */
	name?: string;
	/**
	 * Whether the checkbox is required.
	 */
	isRequired?: boolean;
}

const {
	size = 'md',
	color = 'default',
	defaultChecked,
	disabled,
	name = generateID('checkbox'),
	label,
	isRequired,
} = Astro.props;

const iconSizes = {
	sm: 14,
	md: 16,
	lg: 24,
};
---
<label 
  class="sui-checkmark-label"
  for={name}
  class:list={[
    disabled && "disabled",
    color,
    size,
  ]}
>
  <div
    class="sui-checkmark-container"
    tabindex="0"
    role="checkbox"
    aria-checked={defaultChecked} 
    aria-labelledby={`label-${name}`}
    >
    <Checkmark
      class={'sui-checkmark'}
      width={iconSizes[size]}
      height={iconSizes[size]}
    />
    <input
      type="checkbox"
      name={name}
      id={name}
      checked={defaultChecked}
      disabled={disabled}
      required={isRequired}
      class="sui-checkbox"
      hidden
    />
  </div>
  <span id={`label-${name}`}>
    {label} <span class="req-star">{isRequired && "*"}</span>
  </span>
</label>
<script>
import "studiocms:ui/scripts/checkbox"
</script>
