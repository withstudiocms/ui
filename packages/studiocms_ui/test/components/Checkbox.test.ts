import { describe, expect } from 'vitest';
import Checkbox from '../../src/components/Checkbox/Checkbox.astro';
import { test } from '../fixtures/vitest/AstroContainer';

const recurringProps = {
	label: 'Checkbox Label',
	name: 'checkbox-test',
};

describe('Checkbox Component', () => {
	test('renders Checkbox component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(Checkbox, 'Checkbox', {
			props: { ...recurringProps },
		});
		expect(result).toMatchSnapshot();
	});

	test.for([
		{ ...recurringProps, defaultChecked: true },
		{ ...recurringProps, disabled: true },
		{ ...recurringProps, size: 'sm' },
		{ ...recurringProps, size: 'md' },
		{ ...recurringProps, size: 'lg' },
		{ ...recurringProps, color: 'default' },
		{ ...recurringProps, color: 'primary' },
		{ ...recurringProps, color: 'success' },
		{ ...recurringProps, color: 'warning' },
		{ ...recurringProps, color: 'danger' },
		{ ...recurringProps, color: 'info' },
		{ ...recurringProps, color: 'mono' },
		{ ...recurringProps, isRequired: true },
		{ ...recurringProps, value: 'customValue' },
	])('renders Checkbox with props "%s"', async (props, { renderComponent }) => {
		const result = await renderComponent(Checkbox, 'Checkbox', { props });
		expect(result).toMatchSnapshot();
	});
});
