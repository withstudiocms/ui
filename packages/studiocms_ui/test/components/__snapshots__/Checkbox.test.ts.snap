// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox Component > renders Checkbox component correctly 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'danger' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label danger md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'default' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'info' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label info md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'mono' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label mono md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'primary' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label primary md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'success' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label success md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', color: 'warning' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label warning md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', defaultChecked: true }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-checked="true" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" checked class="sui-checkbox" hidden aria-checked="true"> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', disabled: true }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label disabled default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" disabled class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', isRequired: true }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" required class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star">*</span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', size: 'lg' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default lg"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="24" height="24" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', size: 'md' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', size: 'sm' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default sm"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="14" height="14" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`Checkbox Component > renders Checkbox with props "{ label: 'Checkbox Label', name: 'checkbox-test', value: 'customValue' }" 1`] = `"<label for="checkbox-test" class="sui-checkmark-label default md"> <div class="sui-checkmark-container" tabindex="0" role="checkbox" aria-labelledby="label-checkbox-test"> <svg role="presentation" viewBox="0 0 17 18" width="16" height="16" class="sui-checkmark"> <polyline fill="none" points="1 9 7 14 15 4" stroke="currentColor" stroke-dasharray="22" stroke-dashoffset="66" stroke-linecap="round" stroke-linejoin="round" stroke-width="2"></polyline> </svg> <input type="checkbox" name="checkbox-test" id="checkbox-test" class="sui-checkbox" value="customValue" hidden> </div> <span id="label-checkbox-test"> Checkbox Label <span class="req-star"></span> </span> </label> <script type="module" src="/mock/path/Checkbox.astro?astro&type=script&index=0&lang.ts"></script>"`;
